### ISSUE TITLE
Nemo keeps freezing

### ISSUE THREAD

**Issue by Souravgoswami** [linuxmint/nemo#2605]
```
 * Nemo version 4.8.2
 * Is issue with desktop or windowed nemo? Desktop nemo
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...) Arch. Using XFCE 4.16, but issue is persistent with 4.12, 4.14, and 4.16 (what I have tried so far).
 * Graphics hardware *and* driver used: Intel Corporation Skylake GT2 [HD Graphics 520] (rev 07), i915
 * OS: 64 bit
 ```

**Issue**
I am an old nemo user. I have been using Arch for like 3 years. I always see nemo freezes after some time of use, when I go to various directories that has music and photos and other files, etc. It gets slower and slower to respond as I load more and more things. After a long time, clicking on a folder and moving inside it takes 4 - 5 seconds even it's empty. Loading anything takes a huge time. I have also tried other file managers like nautilus and caja. They never had such problem.

**Steps to reproduce**
Just go to directories with thousands of files, load the thumbnails (generated ones will work, too), and you will see nemo gets slow.

**Comment by ItzSwirlz** [comment]
Do you have any logs or syslogs of nemo?

**Comment by Souravgoswami** [comment]
I don't know how to enable logs. But I am interested in doing that and reproduce this problem.

**Comment by Fantu** [comment]
@Souravgoswami open terminal, kill the nemo process and relaunch from it for see its logs (if will show useful error/warning), it is not certain that give you something related (to this issue) in those logs with freeze (in which case it will be needed a more in-depth debugging)

**Comment by dl9px** [comment]
I also get this failure with nemo 4.8.2 on a fresh 20.1 install. Especially, when my woman, who has a large picture collection, navigates with nemo. I couldn't reproduce the error today and it is also not showing up in system journal. Maybe there is a problem with thumbnails, which are still generated for many pictures.

**Comment by Fantu** [comment]
@dl9px your issue seems different from this issue, thumbnails generation with big number of image, big resolution and/or slow destination can cause delay, if don't cause crash is not an issue, a workaround is enable thumbnails generation only on local fs and limit the size to small flies (see in nemo preferences)

**Comment by Souravgoswami** [comment]
I don't see a lot of messages. The current messages I see are:

```
nemo 

** (nemo:5517): WARNING **: 17:10:54.198: Current gtk theme is not known to have nemo support (XFCE_Colour_Lite_Purple) - checking...

** (nemo:5517): WARNING **: 17:10:54.353: The theme appears to have no nemo support.  Adding some...

(nemo:5517): GLib-GObject-CRITICAL **: 17:11:12.192: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
Gtk-Message: 17:11:15.930: Failed to load module "xapp-gtk3-module"
Gtk-Message: 17:11:15.930: Failed to load module "xapp-gtk3-module"
Gtk-Message: 17:12:15.702: Failed to load module "xapp-gtk3-module"
Gtk-Message: 17:12:15.702: Failed to load module "xapp-gtk3-module"
```

This is 5 minutes prior to the lags, but when it started lagging I see nothing much...

**Comment by Fantu** [comment]
About `Failed to load module "xapp-gtk3-module"` seems something is missed or wrong about xapp, probably is the first thing to try to solve.

**Comment by Souravgoswami** [comment]
Not sure why that happens though. I have xapp installed, and there are the shared objects under /usr/lib/:

```
\ls /usr/lib/libxap*
/usr/lib/libxapian.la  /usr/lib/libxapian.so.30       /usr/lib/libxapp.so    /usr/lib/libxapp.so.2.0.3
/usr/lib/libxapian.so  /usr/lib/libxapian.so.30.10.3  /usr/lib/libxapp.so.1
```

**Comment by Souravgoswami** [comment]
Well, so I had a file `/etc/X11/xinit/xinitrc.d/80xapp-gtk3-module.sh`, I edited it, and removed `xapp-gtk3-module` from there. I then logged out and logged back in. The warning message is gone. But nemo kept freezing. The latest log from nemo is now:

```
** (nemo:6049): WARNING **: 12:32:20.508: Current gtk theme is not known to have nemo support (XFCE_Colour_Lite_Purple) - checking...

** (nemo:6049): WARNING **: 12:32:20.624: The theme appears to have no nemo support.  Adding some...

** (nemo:6049): WARNING **: 12:33:29.672: Could not inhibit power management: GDBus.Error:org.freedesktop.DBus.Error.NameHasNoOwner: Name "org.gnome.SessionManager" does not exist

(nemo:6049): CinnamonDesktop-WARNING **: 12:33:57.644: Unable to create loader for mime type audio/mpeg: Unrecognised image file format

(nemo:6049): CinnamonDesktop-WARNING **: 12:33:57.645: Error creating thumbnail for recent:///39db594e1bcb86ec12ba80a95ff16c5c: Unrecognised image file format

(nemo:6049): Gtk-CRITICAL **: 12:34:25.206: gtk_widget_draw: assertion '!widget->priv->alloc_needed' failed

(nemo:6049): Gtk-CRITICAL **: 12:35:50.325: gtk_widget_destroy: assertion 'GTK_IS_WIDGET (widget)' failed
```

**Comment by Fantu** [comment]
New xapp and nemo version with some fixes is out, probably is good try them (if not already done).
After, about 80xapp-gtk3-module debian for example need a change for support it correctly (see https://salsa.debian.org/cinnamon-team/xapp/-/blob/master/debian/rules),  I don't know about arch

**Comment by SparkyBluefang** [comment]
The `xapp-gtk3-module` warning is coming from GTK2 apps, so it doesn't have anything to do with nemo.


Try disabling all icon captions. `Edit -> Preferences -> Display -> Icon Captions` and setting them all to `None`. Then kill all nemo instances (`killall -9 nemo` or `kill -9 <pids>`). I have found that nemo windows wont pick up the the preference changes if it is already misbehaving, which is why you need to kill it.

On my very large image folders, I can reproduce the problem by opening images as I scroll around. Relatively quickly (within a minute), nemo starts freezing longer and longer when the window gains focus. At least in my case, the freezing seems to be related to scrolling. If I don't scroll (or immediately scroll once upon opening a folder) I can open and close as many images as I want without freezing as long as I don't scroll again.

With the icon captions disabled, I can't reproduce the freezes (at least within a few minutes).

**Comment by Souravgoswami** [comment]
Yup @SparkyBluefang, disabling the Icon captions as you mentioned fixes the issue!!

I have kept using Nemo for hours now. Yes, it looks like the problem is w/ scrolling... Previously I wasn't able to use nemo like 5 minutes!

**Comment by okaestne** [comment]
Thanks for the workaround @SparkyBluefang (and @Souravgoswami for linking to my issue). The captions seem to cause this. I had the "Size" caption enabled. I restarted and changed the caption to "Owner". It causes the same issues. So it's not the caption type, but the caption itself.

**Comment by SparkyBluefang** [comment]
This should be resolved by #2472 in the next release.
